src/main/java/com/garrytrue/repository/IRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.garrytrue.repository;
+
+/**
+ * Created by garrytrue on 12.03.17.
+ */
+public interface IRepository<T> {
```

Comment:
frameworks are using CRUDRepository to describe what operations are supported.
create
read
update
delete

You can use the same naming